src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1222,6 +1222,20 @@ protected NodeModel(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPo
             RaisesModificationEvents = true;
         }
 
+        /// <summary>
+        /// The method returns the assembly name from which the node originated.
+        /// </summary>
+        /// <returns>Assembly Name</returns>
+        internal virtual AssemblyName GetNameOfAssemblyReferencedByNode()
```

Comment:
maybe a better name is `GetAssemblyNameOfNodeModel` - referenced by node sounds like a dependency of the node to me anyway.